src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -1385,7 +1405,19 @@ protected ColumnConverter columnConverterFor(
 
       // Timestamp is a date + time
       case Types.TIMESTAMP: {
-        return rs -> rs.getTimestamp(col, DateTimeUtils.getTimeZoneCalendar(timeZone));
+        return rs -> {
+          Timestamp timestamp = rs.getTimestamp(col, DateTimeUtils.getTimeZoneCalendar(timeZone));
+          switch (tsGranularity) {
+            case TIMESTAMP_GRANULARITY_LONG_NANOS:
+              return DateTimeUtils.toEpochNanos(timestamp);
```

Comment:
If we did this in an enum, we could just have a method on the enum to convert the timestamp, and have this just be a call into `tsGranularity.convert(timestamp)` . WDYT?